### ISSUE TITLE
About the v8.0

### ISSUE THREAD

**Issue by MizaGBF** [MizaGBF/MizaBOT#9]
As stated in the [discord.py changelog](https://discordpy.readthedocs.io/en/stable/whats_new.html), the next major update will have breaking changes.  
As such, I'm planning an overhaul of how the bot works, which will be the 8.0 version.  
For simplicity sake, I'll try to keep the current `config.json` and `save.json` format (worst case scenario, `save.json` will be automatically converted to the new format).  
The cogs, however, will probably change. If you developed your own, don't expect them to work right away.

**Comment by MizaGBF** [comment]
A beta version of the 8.0 is now available.  
Rough roadmap:  
1. Bugfix it as much as possible during June.  
2. Once discord.py 2.0 is released, fix/add whatever need to be changed/added.  
3. Add any new interesting features.  
4. During the July Unite & Fight, fix the ranking scraper if the need arise.

**Comment by MizaGBF** [comment]
I think the whole thing is close to be debugged, I did a ton of fixes and additions during this GW.  
Now we wait discord.py 2.0 to add all the new stuff (slash commands for example...).  
  
In addition, for those who tried to add the bot and saw it disappear from their servers, it's normal: Some bugs slipped by and I either didn't get notified or the bot didn't leave on refusal. So I nuked most new servers who got affected by this bug.  
Also, in case you missed the new note on the readme/help page: **no** the bot isn't free to be added by everyone and everywhere. I got submerged with requests coming from god knows where and that was a huge waste of time to deal with on top of those bugs.    
My plan is still to open it further in the future but not in its current state.
